src/Microsoft.Health.Dicom.Api/Features/Context/DicomRequestContextMiddleware.cs

```diff
@@ -51,8 +52,12 @@ public async Task Invoke(HttpContext context, IDicomRequestContextAccessor dicom
 
             dicomRequestContextAccessor.RequestContext = dicomRequestContext;
 
-            // Call the next delegate/middleware in the pipeline
-            await _next(context);
+            // Adding operation id for this request thread to the log context.
+            using (LogContext.PushProperty("operationId", System.Diagnostics.Activity.Current?.RootId))
```

Comment:
That was exactly my thought as well. I spent some time trying to find a way to get this done without making it serilog-specific but couldn't succeed so made this change for now. We can revisit this in future to make this better, and have it like this for now so that we get the operation id to correlate the logs.